library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608CueBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2014 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer.text.eia608;
+
+import android.text.Layout;
+import android.text.SpannableStringBuilder;
+import com.google.android.exoplayer.text.Cue;
+
+public class Eia608CueBuilder {
+  private SpannableStringBuilder text;
+  private float position;
+  private float line;
+  private int rowIndex;
+
+  public Eia608CueBuilder() {
+    reset();
+  }
+
+  public Eia608CueBuilder(Eia608CueBuilder other) {
+    text = new SpannableStringBuilder(other.text);
+    position = other.position;
+    line = other.line;
+    rowIndex = other.rowIndex;
+  }
+
+  public void reset() {
+    text = null;
+    position = Cue.DIMEN_UNSET;
+    line = Cue.DIMEN_UNSET;
+    rowIndex = 15; // bottom row is the default
+  }
+
+  public Cue build() {
+    return new Cue(text, Layout.Alignment.ALIGN_NORMAL,
+            line, Cue.LINE_TYPE_FRACTION, Cue.ANCHOR_TYPE_START,
+            position, Cue.TYPE_UNSET, Cue.DIMEN_UNSET);
+  }
+
+  public boolean isEmpty() {
+    return (text == null) || (text.length() == 0);
+  }
+
+  public Eia608CueBuilder setText(SpannableStringBuilder aText) {
+    text = aText;
+    return this;
+  }
+
+  public void setRow(int rowIdx) {
+    if ((rowIdx < 1) || (15 < rowIdx)) {
+      this.line = 0.9f;
+      return;
+    }
+
+    rowIndex = rowIdx; // saved for roll-up feature
+
+    // 10% of screen is left of for safety reasons (analog overscan)
+    // the leftover 80% is divided into 15 equal rows. The problem is that the font size
+    // must match the row line height for this, so at the moment, I scale to 90% instead, to avoid
+    // overlap of the borders around the rows.
+    // -1 as the row and column indices are 1 based in the spec
+    this.line = (rowIdx - 1) / 15f * 0.9f + 0.05f;
```

Comment:
This needs minor updates to get closer to the original standard. To fit 15 lines into the safe area, previously I set 15 points (position is the top left corner of a line) evenly in the middle 80% of the screen. But positioning the bottom line exactly to 80% would cause the text of that last line go under its top left corner - effectively outside the intended safe area. 

So let's distribute "80% - font size" evenly: 
```
final float FONT_SIZE_PERCENTAGE = 0.8f / 15f;
this.line = (rowIdx - 1) / 15f * (0.8f - FONT_SIZE_PERCENTAGE) + 0.1f;
```
But this pulls all the lines closer to each other, and the default subtitle rendering might cause overlapping edges with the default font and subtitle background setting. I solved this by reordering the calls in cuePainter:
- Draw all cue backgrounds first
- then Draw all texts
This way there are no issues of any overlaps with the default fonts and settings.